partifact/config.py

```diff
@@ -9,6 +10,12 @@
 
 CONFIG_PATH = "./pyproject.toml"
 URL_TEMPLATE = "https://{code_artifact_domain}-{aws_account}.d.codeartifact.{aws_region}.amazonaws.com/pypi/{code_artifact_repository}"
+# Regular expression pattern to parse the URL if a dash is detected in the normally parsed aws_account.
+# (?P<code_artifact_domain>(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)): matches the code_artifact_domain, allowing alphanumeric characters and dashes
+# (?P<aws_account>\d+): matches the aws_account, allowing one or more digits
+# (?P<aws_region>[a-z0-9-]+): matches the aws_region, allowing lowercase alphanumeric characters and dashes
+# (?P<code_artifact_repository>.*): matches the code_artifact_repository, allowing any character (except a newline)
+URL_PATTERN = r"https://(?P<code_artifact_domain>(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?))-(?P<aws_account>\d+).d.codeartifact.(?P<aws_region>[a-z0-9-]+).amazonaws.com/pypi/(?P<code_artifact_repository>.*)"
```

Comment:
Overall, this looks good - thanks. My preference would be using a single approach (either regex or `parse`) for everything, though.

I think something like
```python
URL_PATTERN = "https://(?P<code_artifact_domain>.*)-(?P<aws_account>\d+).d.codeartifact.(?P<aws_region>[a-z0-9-]+).amazonaws.com/pypi/(?P<code_artifact_repository>.*)"
```
would work for both cases.

Alternatively, we could stick with `parse` and go with a template like
```python
URL_TEMPLATE = "https://{subdomain}.d.codeartifact.{aws_region}.amazonaws.com/pypi/{code_artifact_repository}"
```
and `rsplit` the result.

What do you think?

Reply:
@davidsteiner I think the regex approach would be cleaner and would be my choice moving foward.

I'll update the PR and I'll run the test suite to make sure the newly proposed regex parsing doesn't fail us!

Reply:
Yeah, I like the regex approach. Thanks!